mempool/mempool.go

```diff
@@ -528,16 +527,26 @@ func (cache *mapTxCache) Push(tx types.Tx) bool {
 		// but deleting a non-existent element is fine
 		delete(cache.map_, string(poppedTx))
 		cache.list.Remove(popped)
+		popped.DetachPrev()
+		popped.DetachNext()
 	}
-	cache.map_[string(tx)] = struct{}{}
 	cache.list.PushBack(tx)
+	cache.map_[string(tx)] = cache.list.Back()
 	return true
 }
 
 // Remove removes the given tx from the cache.
 func (cache *mapTxCache) Remove(tx types.Tx) {
 	cache.mtx.Lock()
-	delete(cache.map_, string(tx))
+	stx := string(tx)
+	// TODO: Can we get value and index at the same time?
```

Comment:
nope

Reply:
Aww okay. #postlaunch it may be worth experimenting with a "prehashed input hashmap. (Alot simpler than it sounds) and then we just hash once with the same fast hash.